File: manual/data/python/tests/test_itertools_permutations.py
import unittest
from function_itertools_permutations import PermutationsProcessor


class TestPermutations(unittest.TestCase):

    def setUp(self):
        self.processor = PermutationsProcessor()

    def test_get_permutations_content(self):
        data = "abc"
        result = self.processor.get_permutations(data)
        expected_result = 'abc, acb, bac, bca, cab, cba\nabc: 1, acb: 1, bac: 1, bca: 1, cab: 1, cba: 1'
        self.assertEqual(result, expected_result)

    def test_permutation_duplicates(self):
        data = "aab"
        result = self.processor.get_permutations(data)
        expected_result = 'aab, aab, aba, aba, baa, baa\naab: 2, aba: 2, baa: 2'
        self.assertEqual(result, expected_result)

    def test_invalid_type_error(self):
        data = [1, 2, 3]  # 非字符串输入
        with self.assertRaises(ValueError):
            self.processor.get_permutations(data)

    def test_non_iterable_error(self):
        data = 123  # 非字符串输入
        with self.assertRaises(ValueError):
            self.processor.get_permutations(data)

    def test_permutations_case_sensitivity(self):
        data = "Aa"
        result = self.processor.get_permutations(data)
        expected_result = 'Aa, aA\nAa: 1, aA: 1'  # 调整预期值为字符串格式
        print(result)
        self.assertEqual(result, expected_result)

if __name__ == '__main__':
    unittest.main()

File: manual/data/java/tests/test_itertools_permutations.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_itertools_permutations {
    private function_itertools_permutations processor;

    @BeforeEach
    void setUp() {
        processor = new function_itertools_permutations();
    }

    @Test
    void test_get_permutations_content() {
        String data = "abc";
        String result = processor.get_permutations(data);
        String expectedResult = "abc, acb, bac, bca, cab, cba\nabc: 1, acb: 1, bac: 1, bca: 1, cab: 1, cba: 1";
        assertEquals(expectedResult, result);
    }

    @Test
    void test_permutation_duplicates() {
        String data = "aab";
        String result = processor.get_permutations(data);
        String expectedResult = "aab, aab, aba, aba, baa, baa\naab: 2, aba: 2, baa: 2";
        assertEquals(expectedResult, result);
    }

    @Test
    void test_invalid_type_error() {
        String data = null;  // 模拟非String类型输入
        assertThrows(IllegalArgumentException.class, () -> {
            processor.get_permutations(data);
        });
    }

    @Test
    void test_non_iterable_error() {
        assertThrows(IllegalArgumentException.class, () -> {
            // 在Java中，我们通过传入null来模拟Python中的非字符串输入
            processor.get_permutations(null);
        });
    }

    @Test
    void test_permutations_case_sensitivity() {
        String data = "Aa";
        String result = processor.get_permutations(data);
        String expectedResult = "Aa, aA\nAa: 1, aA: 1";
        assertEquals(expectedResult, result);
    }
}